Core/src/Material/SurfaceMaterialMapper.cpp

```diff
@@ -279,12 +287,71 @@ void Acts::SurfaceMaterialMapper::mapMaterialTrack(
       ++rmIter;
       continue;
     }
-    if (sfIter != mappingSurfaces.end() - 1 &&
-        (rmIter->position - sfIter->position).norm() >
-            (rmIter->position - (sfIter + 1)->position).norm()) {
-      // Switch to next assignment surface
-      ++sfIter;
+    // Do we need to switch to next assignment surface ?
+    if (sfIter != mappingSurfaces.end() - 1) {
+      int mappingType = sfIter->surface->surfaceMaterial()->mappingType();
+      int nextMappingType =
+          (sfIter + 1)->surface->surfaceMaterial()->mappingType();
```

Comment:
I see, the idea here is to check the current & next mapping type and on basis of this decide where to assign the hit.

I think that's a good idea.

Reply:
That is exactly the idea, if we know the current position and the type of surface for the current and next surface we can decide if we want to change to the next surface or not.